### DISCUSSION TITLE
Device flow qrcode

### DISCUSSION THREAD

**Discussion by trumanbutter** [panva/openid-client#629] [Q&A] [answered]
Calling `client.deviceAuthorization(params)` we're successfully resolving the promise which is great. 
When we log the response to the console we can see a bunch of key value pairs 
``` 
DeviceFlowHandle {
  device_code: ''
  expires_in: 300,
  interval: 5,
  user_code: '',
  verification_uri: '',
  verification_uri_complete: '',
  verification_uri_complete_qrcode: ''
}
```

The one I'm curious about is `verification_uri_complete_qrcode` which returns a base64encoded string. When I try to reference it directly it's undefined.
Looking through the DeviceFlowHandle class -> `index.d.ts` there's no reference to `verification_uri_complete_qrcode`

```
export class DeviceFlowHandle<TClient extends BaseClient = BaseClient> {
  poll(options?: DeviceFlowPollOptions): Promise<TokenSet>;
  abort(): void;
  expired(): boolean;
  expires_at: number;
  client: TClient;
  user_code: string;
  device_code: string;
  verification_uri: string;
  verification_uri_complete: string;
  expires_in: number;
}

Does openid-client support this value?

**Comment by panva** [comment, marked as answer]
It's a proprietary, non-standard response property, it cannot be a part of the type definition.

**Reply by trumanbutter** [reply]
Ok, I see that makes sense.
